src/shared/sam/localLambdaRunner.ts

```diff
@@ -303,6 +303,16 @@ export async function invokeLambdaFunction(
             })
     } else {
         // 'target=code' or 'target=template'
+        // Compile step for TS projects from source code
+        if (config.invokeTarget.target === 'code' && (await readdir(config.codeRoot)).includes('tsconfig.json')) {
+            try {
+                getLogger('channel').info('Compiliing TypeScript')
+                await new ChildProcess(true, `tsc --project ${samBuildOutputFolder}\\awsToolkitSamLocalResource --inlineSourceMap`).run()    
+            } catch (error) {
+                getLogger('channel').error(`Compile Error: ${error}`)
+            }
+        }
```

Comment:
There are no runtime-specific workarounds in localLambdaRunner.ts. Let's maintain that.

Current architecture is that runtime-specific run/debug logic lives in `src/shared/sam/debugger/*SamDebug.ts`. There is already a `onAfterBuild` argument. That looks like a good place to put logic like this...

I don't know why this PR has done this step _after_ "sam build", but if that is correct then it could be done in `onAfterBuild` called from `typescriptSamDebug.ts`.

Reply:
I previously had this step before the "sam build", but the files were deleted in the sam build step.  I'll move this logic to ```onAfterBuild```